enclave/core/sgx/calls.c

```diff
@@ -159,6 +159,10 @@ static oe_result_t _eeid_patch_memory()
 
         // Wipe the marker page
         memset(heap_base, 0, OE_PAGE_SIZE);
+
+        // Re-initialize the memory allocator with updated heap size
+        oe_allocator_init(
```

Comment:
@wintersteiger Thanks for identifying the issue.

It is not safe to call `oe_allocator_init` here since we will end up calling it again when the enclave thread is initialized.
https://github.com/openenclave/openenclave/blob/90ab2aa8aec6768d9c9b0208073b8d9cdb8f6933/enclave/core/sgx/threadlocal.c#L377

It seems to me that patching the enclave memory is better done as one of the first things during enclave initialization.
A good place to do that would be here: https://github.com/openenclave/openenclave/blob/e90dd1268a8771c8ccc7de360d74d7b745f7d741/enclave/core/sgx/init.c#L47

Reply:
Yeah, it's intended to be done as the first thing. I'll take another look and move it to the place you suggested.

Reply:
Done! Thanks for the suggestion again, I think this is definitely a better place to do this.